src/qt/rpcconsole.cpp

```diff
@@ -612,9 +615,14 @@ void RPCConsole::setClientModel(ClientModel *model, int bestblock_height, int64_
         ui->peerWidget->setSelectionBehavior(QAbstractItemView::SelectRows);
         ui->peerWidget->setSelectionMode(QAbstractItemView::ExtendedSelection);
         ui->peerWidget->setContextMenuPolicy(Qt::CustomContextMenu);
-        ui->peerWidget->setColumnWidth(PeerTableModel::Address, ADDRESS_COLUMN_WIDTH);
-        ui->peerWidget->setColumnWidth(PeerTableModel::Subversion, SUBVERSION_COLUMN_WIDTH);
-        ui->peerWidget->setColumnWidth(PeerTableModel::Ping, PING_COLUMN_WIDTH);
+
+        QSettings settings;
+        m_peer_widget_header_state = settings.value("PeersTabPeerHeaderState").toByteArray();
```

Comment:
Load `m_peer_widget_header_state` in constructor where `settings` already exists.

Reply:
@promag 

Thanks for raising this question!

From my [experience](https://github.com/bitcoin-core/gui/pull/205#issuecomment-807812020) `QTableView::setColumnWidth` does not work without model set (but it is not documented by Qt).

Did I miss something? Does it work for you?

Reply:
I mean move L620 to constructor but leave `restoreState()` here.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/256#issuecomment-829545241).